### ISSUE TITLE
Handyman teleports across the hospital to fix machine

### ISSUE THREAD

**Issue by tobylane** [CorsixTH/CorsixTH#1013]
#### Describe the issue

In the x-ray (in the east building) the handyman appears from nowhere. I note his salary and reload the savegame. The handyman earning $91 (third on staff list) leaves the operating theatre and teleports to the x-ray to fix it
#### Steps to Reproduce
1. _Load save game attached to this ticket_
2. Before April 8th find the handyman going from reception to operating theatre and open the dialog that follows him.
3. Same thing happens a few times over, with no repair job completed and no damage taken by further x-ray use.
#### Save Game

http://cl.ly/0C2u3K2X322c
#### Expected Behavior

He uses his feet to transport himself to the x-ray room, or a fellow handyman does the job.
#### System Information

**CorsixTH Version:**
d1f9f9e8 
**Operating System:**
OS X 10.10.3
**Theme Hospital Version:**
CD
#### Gamelog.txt

No output

**Comment by mugmuggy** [comment]
Somehow the handyman_position of the x-ray machine didn't match the standard use_position. Its set as (1,0) instead of what I expect would be (-1,1). The (1,0) leads to a non-passable tile and the pathfinder I assume aborts early in the code with !pMap->getNodeUnchecked(iEndX, iEndY)->flags.passable) as passable is false. The handyman is then teleported rather than walking there as the tile positions are updated still. Rebuilding the room corrects the problem.

**Comment by lewri** [comment]
Potential duplicate of #473

**Comment by ARGAMX** [comment]
@tobylane do you have savegame of this issue?

**Comment by tobylane** [comment]
No, and if I understand Muggy's comment, it can't be genuinely recreated with that faulty use_position.

The use_position should be reachable, and if rebuilding the room fixes this, then that may be fixed. Second, should the path finding create as complete a path as it can, ie to right before the unpassable tile, on a different side of the machine, and have the animation start there?

**Comment by ARGAMX** [comment]
> No, and if I understand Muggy's comment, it can't be genuinely recreated with that faulty use_position.
> 
> The use_position should be reachable, and if rebuilding the room fixes this, then that may be fixed. Second, should the path finding create as complete a path as it can, ie to right before the unpassable tile, on a different side of the machine, and have the animation start there?

Can this issue can be reproduced in 0.68.0 without `ALLOW BLOCKING OFF AREAS` enabled?

**Comment by Alberth289346** [comment]
> Can this issue can be reproduced in 0.68.0 without ALLOW BLOCKING OFF AREAS enabled?

What's your goal? Do you intend to keep that flag enabled forever?

**Comment by Alberth289346** [comment]
If you don't want to work on issues that are not present in the current release, why not just skip the issues that use that flag?
